addons/html_editor/static/src/others/embedded_components/plugins/toggle_plugin/toggle_plugin.js

```diff
@@ -0,0 +1,504 @@
+                this.dependencies.history.addStep();
+                window.setTimeout(cursors.restore, "animationFrame"); // Used to handle caret displaying issues
+            }
+            return true;
+        }
+    }
+
+    /**
+     * Handles the shift-tab behavior. This means that we need to outdent the toggle from each other.
+     * @returns
+     */
+    handleShiftTab() {
+        const selection = this.dependencies.selection.getEditableSelection();
+        const closestToggleTitle = closestElement(selection.anchorNode, titleSelector);
+        if (closestToggleTitle) {
+            // We are indeed inside a title.
+            const container = closestToggleTitle.closest("[data-embedded]");
+            if (container.parentElement.closest("[data-embedded='toggle']")) {
+                // If we are inside an indented toggle we need to outdent the current toggle.
+                const nextPosition = rightPos(
+                    container.parentElement.closest("[data-embedded='toggle']")
+                );
+                const cursors = this.dependencies.selection.preserveSelection();
+                this.dependencies.selection.setSelection({
+                    anchorNode: nextPosition[0],
+                    anchorOffset: nextPosition[1],
+                    focusNode: nextPosition[0],
+                    focusOffset: nextPosition[1],
+                });
+                this.dependencies.dom.insert(container);
+                cursors.restore();
+                this.dependencies.history.addStep();
+            }
+            return true;
+        }
+    }
+
+    handleSplitElementBlock({ targetNode }) {
+        if (targetNode.closest("[data-embedded='toggle'] [data-embedded-editable='title']")) {
+            const selection = this.dependencies.selection.getEditableSelection();
+            if (isEmptyBlock(selection.anchorNode)) {
+                // If no text is in title, we remove the toggle.
+                const newParagraph = this.document.createElement("p");
+                newParagraph.appendChild(this.document.createElement("br"));
+                targetNode.closest("[data-embedded='toggle']").replaceWith(newParagraph);
+                this.dependencies.selection.setCursorStart(newParagraph);
+                return true;
+            }
+            let insertBefore;
+            const container = targetNode.closest("[data-embedded='toggle']");
+            const insertInside = container.firstChild.querySelector(".fa-caret-down");
+            const anchorNode = selection.anchorNode.previousSibling ?? selection.anchorNode;
+            if (selection.isCollapsed && selection.endOffset === 0) {
+                insertBefore = selection.anchorNode.previousSibling !== null;
+            }
+            if (insertInside) {
+                // Toggle is open
+                const target = container
+                    .querySelector(contentSelector)
+                    .querySelector("*:first-child");
```

Comment:
Isn't this equivalent to this ?
```js
container.querySelector(`${contentSelector} *:first-child`)
```

Reply:
Oh yes I see. Could probably use the `:scope` selector if we really want to avoid the usage of 2 consecutive selectors 😅.
Or even simpler, maybe we can replace the second selector by `.firstElementChild`?